src/components/dataset-query/dataset-query.js

```diff
@@ -26,30 +26,34 @@ const DatasetQuery = props => {
   const submit = () => {
     // I'd like to use the Redux queryText here, but this way makes a test pass -JBP 10/9/2019
     executeQuery(localQueryText)
-    setIsCancelled(false)
+    setUserInteracted()
```

Comment:
Is there a reason this can't be grouped in with the `EXECUTE_QUERY` and `CANCEL_QUERY` events? Using actions/events as getters and setters on single pieces of redux state (whether intentional or not) is a common pitfall and generally considered an anti-pattern.

I know it makes some sense, since it reflects the state we're consuming back in (`hasUserInteracted`). However, it might be better just to keep that state completely hidden and instead expose the `showSuccessMessage` and `showFailureMessage` stuff through a selector (which should be an okay place for that IMO)

Reply:
For your first question, we definitely could; but that would mean passing in an additional boolean to `executeFreestyleQuery` to indicate the user initiated it rather than it being an auto-load. I started down that path but this way seemed slightly simpler.

For the second, we can definitely do that now that all of this is in Redux.

Reply:
Aha, I gotcha